src/Data/Digit.agda

```diff
@@ -87,20 +90,32 @@ toDigits base@(suc (suc k)) n = <′-rec Pred helper n
   cons : ∀ {m} (r : Digit base) → Pred m → Pred (toℕ r + m * base)
   cons r (ds , eq) = (r ∷ ds , cong (λ i → toℕ r + i * base) eq)
 
-  open ≤-Reasoning
-  open +-*-Solver
+  lem′ : ∀ x k → x + x + (k + x * k) ≡ k + x * 2+ k
+  lem′ x k = begin
+    x + x + (k + x * k)   ≡⟨ +-assoc (x + x) k _ ⟨
+    x + x + k + x * k     ≡⟨ cong (_+ x * k) (+-comm _ k) ⟩
+    k + (x + x) + x * k   ≡⟨ +-assoc k (x + x) _ ⟩
+    k + ((x + x) + x * k) ≡⟨ cong (k +_) (begin
+      (x + x) + x * k       ≡⟨ +-assoc x x _ ⟩
+      x + (x + x * k)       ≡⟨ cong (x +_) (cong (_+ x * k) (*-identityʳ x)) ⟨
+      x + (x * 1 + x * k)   ≡⟨ cong₂ _+_ (*-identityʳ x) (*-distribˡ-+ x 1 k ) ⟨
+      x * 1 + (x * suc k)   ≡⟨ *-distribˡ-+ x 1 (1 + k) ⟨
+      x * 2+ k              ∎) ⟩
```

Comment:
The alignment of the LHSs got messed up here.

Reply:
Actually, this is intentional -- this is a nested proof, see the 'begin' on the line above.